### PR TITLE
Refactor hydrus

### PR DESCRIPTION
**Pull request by sameshl**
related to #424 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

* refactor error responses with `error_response` function
* refactor validation logic of objects while inserting / updating using `validate_object` function
* replace  `.format(...)` strings with Python 3.6+'s "f-strings" in the whole hydrus codebase. Not only are f-strings more readable, more concise, and less prone to error than other ways of formatting, but they are also faster.

### CODE REVIEW

**Comment by sameshl** [comment]
@xadahiya The build is failing for python version < 3.6, where f-strings were not introduced. Do you still want to keep the changes? I thought f strings would be a great improvement to the codebase in terms of readability/maintainability.

**Comment by Mec-iS** [comment]
> @xadahiya The build is failing for python version < 3.6, where f-strings were not introduced. Do you still want to keep the changes? I thought f strings would be a great improvement to the codebase in terms of readability/maintainability.

I am also starting to think that we may drop compatibility of Python 3.5, as for now everybody is on the move to 3.6 and we can easily do it at this stage of adoption

**Comment by sameshl** [comment]
> I am also starting to think that we may drop compatibility of Python 3.5, as for now everybody is on the move to 3.6 and we can easily do it at this stage of adoption

Yes, exactly.

**Comment by xadahiya** [comment]
The build is failing. @sameshl, please check

**Comment by sameshl** [comment]
@xadahiya As mentioned earlier the build is failing only for python < 3.6 as I have used f strings while refactoring.

**Comment by xadahiya** [comment]
Okay, I think there's no harm in dropping support for python < 3.6. Can you update travis config and README and related docs? Then we can merge

**Comment by sameshl** [comment]
> Okay, I think there's no harm in dropping support for python < 3.6. Can you update travis config and README and related docs? Then we can merge

Yes. Will do

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,8 @@
 language: python
 python:
-  - "3.5"
-  - "3.5-dev"  # 3.5 development branch
   - "3.6"
-  - "3.6-dev"  # 3.6 development branch
```
Comment (xadahiya) [inline .travis.yml]: Why did you remove 3.6-dev and 3.7-dev?

Comment (sameshl) [reply .travis.yml]: Ah. I thought we would not need to test for the dev branches if we are building against python 3.8.

Comment (sameshl) [reply .travis.yml]: Will revert the change.

**Comment by sameshl** [comment]
@xadahiya I have made the required changes. Could you have a look?
